### PR TITLE
Add ASAN and TSAN integration testing

### PR DESCRIPTION
**Pull request by ncesario-lunarg**
Add jobs for testing with -fsanitize=address and -fsanitize=thread.

### CODE REVIEW

File: StandAlone/StandAlone.cpp [hunk 1]
```diff
@@ -144,8 +145,9 @@ void FreeFileData(char* data);
 void InfoLogMsg(const char* msg, const char* name, const int num);
 
 // Globally track if any compile or link failure.
-bool CompileFailed = false;
-bool LinkFailed = false;
+std::atomic<int8_t> CompileFailed{0};
```
Comment (arcady-lunarg) [inline StandAlone/StandAlone.cpp:148]: Would it be better to use `std::atomic<bool>` here, or is there some reason this has to be int8_t?

**Review by arcady-lunarg — COMMENTED** [review]
I just have one question about the atomics, otherwise this looks good, thanks for getting it done.

File: StandAlone/StandAlone.cpp (hunk 1 above)
Comment (ncesario-lunarg) [reply StandAlone/StandAlone.cpp:148]: `std::atomic<bool>` would have been my first choice, but it doesn't seem to work with `fetch_or`.

Comment (arcady-lunarg) [reply StandAlone/StandAlone.cpp:148]: Ah, okay. That's annoying but I suppose it's not a big deal to use int8_t.

**Review by arcady-lunarg — APPROVED** [review]
LGTM, thanks.

**Pull request merged**